smart-contracts/wasm-chain-integration/src/v1/trie/api.rs

```diff
@@ -146,17 +149,48 @@ impl PersistentState {
         }
     }
 
-    /// Lookup a key in the tree. This is only meant for testing
-    /// since performance is slow compared to lookup in the mutable tree.
+    /// Lookup a key in the tree and return a copy of the value stored.
     pub fn lookup(&self, loader: &mut impl BackingStoreLoad, key: &[u8]) -> Option<Value> {
         match self {
             PersistentState::Empty => None,
-            PersistentState::Root(node) => {
-                let node = node.get(loader);
-                let data = node.data.lookup(loader, key)?;
-                let borrowed = data.borrow();
-                Some(borrowed.get_copy(loader))
+            PersistentState::Root(root_node) => {
+                let mut trie = root_node.make_mutable(0, loader);
```

Comment:
Please add a comment explaining that you make it mutable to increase performance.